### PR TITLE
Fixes #26364 - Copy modules on incremental update

### PR DESCRIPTION
**Pull request by parthaa**
This commit copies modules associated to an errata on an incremental
publish.
Note:
This is likely to get fixed when pulp 2.19 shows up with the dep
solving features. Then the recursive:true to errata copy would
automatically copy the associated  modules. This commit is more of a
stopgap.

### CODE REVIEW

File: app/services/katello/pulp/repository.rb [hunk 1]
```diff
@@ -238,6 +238,18 @@ def copy_units(destination_repo, units, options = {})
...
+          # remove this code after 2.19
+          module_stream_ids = units.map do |erratum|
+            erratum.module_stream_objects.map(&:pulp_id)
+          end
+          module_stream_ids.flatten!
+          smart_proxy.pulp_api.extensions.module.copy(repo.pulp_id,
+                                                      destination_repo.pulp_id,
+                                                      ids: module_stream_ids,
+                                                      override_config: override_config)
+        end
```
Comment (jlsherrill) [inline app/services/katello/pulp/repository.rb]: could you put this in the yum service class?  Maybe override this method and then call super() followed by this code?

Comment (parthaa) [reply app/services/katello/pulp/repository.rb]: you sure thats a better approach ? Considering this is a stopgap

Comment (jlsherrill) [reply app/services/katello/pulp/repository.rb]: I think so and then we can just delete the overridden method?

Comment (parthaa) [reply app/services/katello/pulp/repository.rb]: moreover I am using same override config as errata

Comment (jlsherrill) [reply app/services/katello/pulp/repository.rb]: ah, i guess its fine then, since this is temporary

**Comment by omkarkhatavkar** [comment]
@parthaa with this PR and https://github.com/Katello/katello/pull/8046 I'm not able to copy the modular rpms in content view with non modular pacakge filter? This is getting against the https://github.com/Katello/katello/pull/8014. 

```
hammer> content-view filter list --content-view mytest_content_view_1 --organization-id 2
----------|----------------|------|----------
FILTER ID | NAME           | TYPE | INCLUSION
----------|----------------|------|----------
5         | package filter | rpm  | true     
----------|----------------|------|----------
hammer> content-view filter info --id 5
Filter ID:    5
Name:         package filter
Type:         rpm
Inclusion:    true
Description:  test
Repositories: 

Rules:        
 1) ID:      1
    Name:    squirrel
    Created: 2019/03/28 09:59:23
    Updated: 2019/03/28 09:59:23

```
Versions for Content View  mytest_content_view_1

**Comment by parthaa** [comment]
> @parthaa with this PR and #8046 I'm not able to copy the modular rpms in content view with non modular pacakge filter? This is getting against the #8014.

Yeah 3 different PRs. Code for 8014 has to get merged to master for you to be able to copy modular rpms on publish. This one is testing incremental publish.

**Comment by jlsherrill** [comment]
My steps for testing on master:
1) make sure this pr and https://github.com/Katello/katello/pull/8048 are checked out
2) create a repo and sync http://quartet.usersys.redhat.com/pub/repos/errata_test/modules_1/  (note this url isn't publicly accessible)
3) publish in a cv
4) change url on repository to http://quartet.usersys.redhat.com/pub/repos/errata_test/modules_2/  resync the repo
5) run:
```
hammer content-view version incremental-update --content-view-version-id=51 --errata-ids=RHEA-2013:0098
```
(change the content view id to the 1.0 version of your content view

6) examine the 1.1 content view version for the # of errata and modules, it should be one more of each than 1.0

My results:
image

**Comment by parthaa** [comment]
Bug created -> https://projects.theforeman.org/issues/26478

**Comment by omkarkhatavkar** [comment]
> > @parthaa with this PR and #8046 I'm not able to copy the modular rpms in content view with non modular pacakge filter? This is getting against the #8014.
> 
> Yeah 3 different PRs. Code for 8014 has to get merged to master for you to be able to copy modular rpms on publish. This one is testing incremental publish.

Yeah i missed to add that one

**Comment by omkarkhatavkar** [comment]
works fine same as https://github.com/Katello/katello/pull/8048#issuecomment-477646725

**Pull request merged**